oxenc/bt_producer.h

```diff
@@ -299,19 +334,38 @@ class bt_list_producer {
         return bs->init + next + 1;
     }
 
+    template <string_like T>
+    void append(const T& data) {
+        if (has_child)
+            throw std::logic_error{"Cannot append to list when a sublist is active"};
+        append_impl(data.data(), data.size());
+        append_intermediate_ends();
+    }
+
     /// Appends an element containing binary string data
-    template <string_view_compatible T>
+    template <char_span_convertible T>
     void append(const T& data) {
         if (has_child)
             throw std::logic_error{"Cannot append to list when a sublist is active"};
-        append_impl(data);
+        append_impl(data.data(), data.size());
         append_intermediate_ends();
     }
 
-    template <string_view_compatible T>
-    bt_list_producer& operator+=(const T& data) {
-        append(data);
-        return *this;
+    /// Appends an element containing a const Char string
+    void append(std::string_view s) {
+        if (has_child)
+            throw std::logic_error{"Cannot append to list when a sublist is active"};
+        append_impl(s);
+        append_intermediate_ends();
+    }
+
+    /// Appends an element containing const_span data
+    template <basic_char T>
+    void append(const std::span<T>& data) {
```

Comment:
Should this not be a `std::span<const T> data` instead, so that it is invokable from (for example) a const vector?

Reply:
For that matter: wouldn't a single `std::span<const T> data` be generic enough to *also* handle the string_view case?

Reply:
Oh, never mind, that's wrong: the deduction means this will only match literal spans, not convertible-to-spans.